egui_plot/src/items/span.rs

```diff
@@ -0,0 +1,336 @@
+    #[inline]
+    pub fn border(mut self, stroke: impl Into<Stroke>) -> Self {
+        self.border_stroke = stroke.into();
+        self
+    }
+
+    /// Convenience for updating the span border width.
+    #[inline]
+    pub fn border_width(mut self, width: impl Into<f32>) -> Self {
+        self.border_stroke.width = width.into();
+        self
+    }
+
+    /// Convenience for updating the span border color.
+    #[inline]
+    pub fn border_color(mut self, color: impl Into<Color32>) -> Self {
+        self.border_stroke.color = color.into();
+        self
+    }
+
+    /// Set the style for the span borders. Defaults to `LineStyle::Solid`.
+    #[inline]
+    pub fn border_style(mut self, style: LineStyle) -> Self {
+        self.border_style = style;
+        self
+    }
+
+    #[inline]
+    pub(crate) fn fill_color(&self) -> Color32 {
+        self.fill
+    }
+
+    #[inline]
+    pub(crate) fn border_color_value(&self) -> Color32 {
+        self.border_stroke.color
+    }
+
+    fn range_sorted(&self) -> (f64, f64) {
+        let start = *self.range.start();
+        let end = *self.range.end();
+        if start <= end {
+            (start, end)
+        } else {
+            (end, start)
+        }
+    }
+
+    fn hline_points(value: f64, transform: &PlotTransform) -> Vec<Pos2> {
+        vec![
+            transform.position_from_point(&PlotPoint::new(transform.bounds().min[0], value)),
+            transform.position_from_point(&PlotPoint::new(transform.bounds().max[0], value)),
+        ]
+    }
+
+    fn vline_points(value: f64, transform: &PlotTransform) -> Vec<Pos2> {
+        vec![
+            transform.position_from_point(&PlotPoint::new(value, transform.bounds().min[1])),
+            transform.position_from_point(&PlotPoint::new(value, transform.bounds().max[1])),
+        ]
+    }
+
+    fn draw_border(
+        &self,
+        value: f64,
+        stroke: Stroke,
+        transform: &PlotTransform,
+        shapes: &mut Vec<Shape>,
+    ) {
+        if stroke.color == Color32::TRANSPARENT || stroke.width <= 0.0 {
+            return;
+        }
+
+        let line = match self.axis {
+            Axis::X => Self::vline_points(value, transform),
+            Axis::Y => Self::hline_points(value, transform),
+        };
+
+        self.border_style.style_line(
+            line,
+            PathStroke::new(stroke.width, stroke.color),
+            false,
+            shapes,
+        );
+    }
+
+    fn available_width_for_name(&self, rect: &Rect) -> f32 {
+        match self.axis {
+            Axis::X => (rect.width() - 2.0 * LABEL_PADDING).max(0.0),
+            Axis::Y => (rect.height() - 2.0 * LABEL_PADDING).max(0.0),
+        }
+    }
+
+    // If the span is too small to display the full name, find the longest name
+    // with "..." appended that we can display within the span
+    fn find_name_candidate(&self, width: f32, painter: &Painter, font_id: &FontId) -> String {
```

Comment:
Let's extract this as a standalone util function, it can be used in other places.